falcon/api.py

```diff
@@ -52,66 +52,27 @@ def process_request(self, req, resp):
-            component modifies ``req.uri`` in its *process_request* method,
-            the framework will use the modified value to route the request.
-
-            Each component's *process_request* and *process_response* methods
-            are executed hierarchically, as a stack. For example, if a list of
-            middleware objects are passed as ``[mob1, mob2, mob3]``, the order
-            of execution is as follows::
-
-                mob1.process_request
-                    mob2.process_request
-                        mob3.process_request
-                            <route to responder method>
-                        mob3.process_response
-                    mob2.process_response
-                mob1.process_response
-
-            Note that each component need not implement both process_*
-            methods; in the case that one of the two methods is missing,
-            it is treated as a noop in the stack. For example, if ``mob2`` did
-            not implement *process_request* and ``mob3`` did not implement
-            *process_response*, the execution order would look
-            like this::
-
-                mob1.process_request
-                    _
-                        mob3.process_request
-                            <route to responder method>
-                        _
-                    mob2.process_response
-                mob1.process_response
-
-            If one of the *process_request* middleware methods raises an
-            error, it will be processed according to the error type. If
-            the type matches a registered error handler, that handler will
-            be invoked and then the framework will begin to unwind the
-            stack, skipping any lower layers. The error handler may itself
-            raise an instance of HTTPError, in which case the framework
-            will use the latter exception to update the *resp* object.
-            Regardless, the framework will continue unwinding the middleware
-            stack. For example, if *mob2.process_request* were to raise an
-            error, the framework would execute the stack as follows::
-
-                mob1.process_request
-                    mob2.process_request
-                        <skip mob3 and routing>
-                    mob2.process_response
-                mob1.process_response
-
-            Finally, if one of the *process_response* methods raises an error,
-            or the routed on_* responder method itself raises an error, the
-            exception will be handled in a similar manner as above. Then,
-            the framework will execute any remaining middleware on the
-            stack.
+                    def process_response(self, req, resp, resource)
+                        \"""Post-processing of the response (after routing).
 
+                        Args:
+                            req: Request object that will eventually be
```

Comment:
At this point, the request was routed to a responder, if one could be found. But even if no route is found, the middleware component's `process_response` method will still be called with the usual Request and Response objects. Rather than trying to explain all this, it may be best to keep things simple in the docstring, e.g.:

```
req: Request object
resp: Response object
```